CT-1553 rename categories to correspondence_types; add JSON params

misaka opened the pull request:
This commit packages together a few changes which weren't straight-forward to
pick apart:

- renaming from categories to correspondence_types, done because we now realise
  what these "categories" are really called (we're calling them Correspondence
  Types for the user), and since we're making other changes to the table to
  support different deadline calculation, we've included this slightly painful
  refactoring in here now
- conversion of attributes from columns to JSON parameters; done to support
  different calculation of deadlines for SAR cases which calculate using
  calendar days and starting from the date of receipt and not the next day after
- addition of a correspondence_types association to BusinessUnit; notable
  because it replaces a bunch of code we had to manage this association
  ourselves, but which also ended up being a bit tricky to get right

misaka commented on config/locales/en.yml:
```diff
@@ -64,7 +65,11 @@ en:
+  alerts:
+    escalation_warning_html: |
+      <strong>Important notice</strong><br>
+      This case in being reviewed by Press and Private office.
```
Small typo ... "This case *in* being reviewed ...", should be _is_ I guess?

aliuk2012 replied:
Good catch 👍

misaka commented on app/views/cases/show.html.slim:
```diff
@@ -1,5 +1,8 @@
-= render partial: 'layouts/header', locals: {page_heading: 'Question'}
+- if @case.triggerable?
```
Is this the only criteria for showing this message? What if we're past the "Day 6" date?

aliuk2012 replied:
Good point, I thought oh just adding this in so its not displayed for other case types. Having spoken to Kellie she agrees that we should hide this if the day has gone past.

EdwardAndress commented on app/views/cases/show.html.slim:
```diff
@@ -1,5 +1,8 @@
-= render partial: 'layouts/header', locals: {page_heading: 'Question'}
+- if @case.triggerable? && Time.now < @case.escalation_deadline
```
Maybe we should have another method on Case that can be used here.

For example

def under_review?
  triggerable? && within_escalation_deadline?
end

def within_escalation_deadline?
  escalation_deadline.future? || escalation_deadline.present?
end

misaka committed:
CT-1553 rename categories to correspondence_types; add JSON params

This commit packages together a few changes which weren't straight-forward to
pick apart:

- renaming from categories to correspondence_types, done because we now realise
  what these "categories" are really called (we're calling them Correspondence
  Types for the user), and since we're making other changes to the table to
  support different deadline calculation, we've included this slightly painful
  refactoring in here now
- conversion of attributes from columns to JSON parameters; done to support
  different calculation of deadlines for SAR cases which calculate using
  calendar days and starting from the date of receipt and not the next day after
- addition of a correspondence_types association to BusinessUnit; notable
  because it replaces a bunch of code we had to manage this association
  ourselves, but which also ended up being a bit tricky to get right
